mod/discussions/start.php

```diff
@@ -0,0 +1,582 @@
+
+	elgg_register_library('elgg:discussion', elgg_get_plugins_path() . 'discussions/lib/discussion.php');
+
+	elgg_extend_view('css/elgg', 'css/discussion');
+
+	elgg_register_page_handler('discussion', 'discussion_page_handler');
+
+	elgg_register_plugin_hook_handler('entity:url', 'object', 'discussion_set_topic_url');
+
+	// commenting not allowed on discussion topics (use a different annotation)
+	elgg_register_plugin_hook_handler('permissions_check:comment', 'object', 'discussion_comment_override');
+	elgg_register_plugin_hook_handler('permissions_check', 'object', 'discussion_can_edit_reply');
+
+	// discussion reply menu
+	elgg_register_plugin_hook_handler('register', 'menu:entity', 'discussion_reply_menu_setup');
+
+	// allow non-owners to add replies to discussion
+	elgg_register_plugin_hook_handler('container_permissions_check', 'object', 'discussion_reply_container_permissions_override');
+
+	elgg_register_event_handler('update:after', 'object', 'discussion_update_reply_access_ids');
+
+	$action_base = elgg_get_plugins_path() . 'discussions/actions/discussion';
+	elgg_register_action('discussion/save', "$action_base/save.php");
+	elgg_register_action('discussion/delete', "$action_base/delete.php");
+	elgg_register_action('discussion/reply/save', "$action_base/reply/save.php");
+	elgg_register_action('discussion/reply/delete', "$action_base/reply/delete.php");
+
+	// add link to owner block
+	elgg_register_plugin_hook_handler('register', 'menu:owner_block', 'discussion_owner_block_menu');
+
+	// Register for search.
+	elgg_register_entity_type('object', 'discussion');
+	elgg_register_plugin_hook_handler('search', 'object:discussion', 'discussion_search_discussion');
+
+	// because replies are not comments, need of our menu item
+	elgg_register_plugin_hook_handler('register', 'menu:river', 'discussion_add_to_river_menu');
+
+	// add the forum tool option
+	add_group_tool_option('forum', elgg_echo('groups:enableforum'), true);
+	elgg_extend_view('groups/tool_latest', 'discussion/group_module');
+
+	$discussion_js_path = elgg_get_site_url() . 'mod/discussions/views/default/js/discussion/';
+	elgg_register_js('elgg.discussion', $discussion_js_path . 'discussion.js');
+
+	elgg_register_ajax_view('ajax/discussion/reply/edit');
+
+	// notifications
+	elgg_register_plugin_hook_handler('get', 'subscriptions', 'discussion_get_subscriptions');
+	elgg_register_notification_event('object', 'discussion');
+	elgg_register_plugin_hook_handler('prepare', 'notification:create:object:discussion', 'discussion_prepare_notification');
+	elgg_register_notification_event('object', 'discussion_reply');
+	elgg_register_plugin_hook_handler('prepare', 'notification:create:object:discussion_reply', 'discussion_prepare_reply_notification');
+
+	// allow ecml in discussion
+	elgg_register_plugin_hook_handler('get_views', 'ecml', 'discussion_ecml_views_hook');
+}
+
+/**
+ * Discussion page handler
+ *
+ * URLs take the form of
+ *  All topics in site:    discussion/all
```

Comment:
i can get the confusion. There is no line in singular/plural page handlers

singular: blog/all, file/all
plural: groups/all, pages/all